### DISCUSSION TITLE
[Modded-Docker-Server] Is there a way to roll the server back to previous stable?

### DISCUSSION THREAD

**Discussion by JohanVlugt** [mbround18/valheim-docker#737] [Help!] [answered]
Hello,

I tried to play today and my modded client gave errors on startup. I rolled back my client via steam but now my server is on a higher version than my client so I can't log in.

Maybe a feature suggestion:
In the ENV let us choose the server version and mod version (ValheimPlus, BepInExPack).

Anyway is it possible to roll back the server? Thanks

**Comment by mbround18** [comment, marked as answer]
Great suggestion! you can in fact do this today! :) typically i highly recommend maintaining backups of your server but
https://www.reddit.com/r/Steam/comments/611h5e/guide_how_to_download_older_versions_of_a_game_on/

following that guide you should be able to, inside the container, download an older version of your server and then all you need to do is restart with an overridden `BEPINEX_DOWNLOAD_URL` to point to the bepinex version you want. 

valheim plus is depreciated by the developer and no longer supported. 

Please let us know how it goes! :)

**Comment by JohanVlugt** [comment]
I managed to downgrade the server but it's still kind of a mess concerning with the server folder. I don't know if these mixed files can corrupt worlds/characters. Anyway I have a backup of at least that.

I downloaded this file in the steam console `download_depot 896660 896661 3933863631502163895` and copied it over to the server.
Then I downloaded the previous bepinex and overwrote the installed one by default.

Started up the server and saw a bunch of logs I have not seen before but it loaded all my mods correctly.
I could join my server without any problem.

Lesson learned, I need to backup my whole server and not set the auto update. Thanks for the advice and this project!

Is there any change those server + bepinex version variables will be exposed in the ENV in future updates?

**Reply by mbround18** [reply]
Potentially, its something that could theoretically be added to Odin as a command. I have a few things going on in my personal life ATM before I'd dive into a feature improvement like that
